src/main/java/org/spongepowered/common/registry/builtin/supplier/EventContextKeySupplier.java

```diff
@@ -0,0 +1,73 @@
+package org.spongepowered.common.registry.builtin.supplier;
+
+import com.google.common.reflect.TypeToken;
+import org.spongepowered.api.CatalogKey;
+import org.spongepowered.api.block.BlockSnapshot;
+import org.spongepowered.api.data.type.HandType;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.entity.living.player.User;
+import org.spongepowered.api.event.block.ChangeBlockEvent;
+import org.spongepowered.api.event.cause.EventContextKey;
+import org.spongepowered.api.event.cause.entity.damage.DamageType;
+import org.spongepowered.api.event.cause.entity.damage.source.DamageSource;
+import org.spongepowered.api.event.cause.entity.dismount.DismountType;
+import org.spongepowered.api.event.cause.entity.spawn.SpawnType;
+import org.spongepowered.api.event.cause.entity.teleport.TeleportType;
+import org.spongepowered.api.item.inventory.ItemStackSnapshot;
+import org.spongepowered.api.plugin.PluginContainer;
+import org.spongepowered.api.profile.GameProfile;
+import org.spongepowered.api.projectile.source.ProjectileSource;
+import org.spongepowered.api.service.ServiceManager;
+import org.spongepowered.api.world.LocatableBlock;
+import org.spongepowered.api.world.World;
+import org.spongepowered.common.event.SpongeEventContextKey;
+
+import java.util.stream.Stream;
+
+@SuppressWarnings("UnstableApiUsage")
+public class EventContextKeySupplier {
+
+    public static Stream<EventContextKey<?>> stream() {
```

Comment:
@Zidane if you can, please tell me this is the correct way of doing it for Sponge provided types?

Reply:
This would go into the Stream package and be named like the other stream classes are.